src/main/java/interview/Parser.java

```diff
@@ -0,0 +1,52 @@
+package interview;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * This class is thread safe.
+ */
+public class Parser {
```

Comment:
It turns out that this class does lots of useful things but not parsing. Don't you thing that it would be better to rename the class to something like `FileContent`?